### ISSUE TITLE
DI Compilation Fails - MageWorx\SeoBase\Helper\StoreUrl

### ISSUE THREAD

**Issue by jbrada** [customgento/module-default-store-code-remover-m2#12]
Hello,

There is an issue in the DI compilation process.

The module fails with the following error message due to MageWorx\SeoBase\Helper\StoreUrl.

```
Compilation was started.
%message% 0/9 [>---------------------------]   0% < 1 sec 105.0 MiBProxies code generation... 0/9 [>-----------------------]   0% < 1 sec 105.0 MiB
Proxies code generation... 1/9 [==>---------------------]  11% 1 sec 109.0 MiB
Repositories code generation... 1/9 [==>------------------]  11% 1 sec 109.0 MiB
There is an error in /data/vendor/customgento/module-default-store-code-remover-m2/Helper/StoreUrl.php at line: 11
Class "MageWorx\SeoBase\Helper\StoreUrl" not found#0 /data/vendor/composer/ClassLoader.php(576): include()
#1 /data/vendor/composer/ClassLoader.php(427): Composer\Autoload\{closure}('/data/vendor/co...')
#2 [internal function]: Composer\Autoload\ClassLoader->loadClass('CustomGento\\Def...')
#3 /data/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(134): class_exists('CustomGento\\Def...')
#4 /data/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(117): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->includeClass('CustomGento\\Def...', '/data/vendor/cu...')
#5 /data/setup/src/Magento/Setup/Module/Di/Code/Reader/ClassesScanner.php(87): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->extract(Object(RecursiveIteratorIterator))
#6 /data/setup/src/Magento/Setup/Module/Di/App/Task/Operation/RepositoryGenerator.php(61): Magento\Setup\Module\Di\Code\Reader\ClassesScanner->getList('/data/vendor/cu...')
#7 /data/setup/src/Magento/Setup/Module/Di/App/Task/Manager.php(56): Magento\Setup\Module\Di\App\Task\Operation\RepositoryGenerator->doOperation()
#8 /data/setup/src/Magento/Setup/Console/Command/DiCompileCommand.php(209): Magento\Setup\Module\Di\App\Task\Manager->process(Object(Closure), Object(Closure))
#9 /data/vendor/symfony/console/Command/Command.php(326): Magento\Setup\Console\Command\DiCompileCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 /data/vendor/symfony/console/Application.php(1078): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /data/vendor/symfony/console/Application.php(324): Symfony\Component\Console\Application->doRunCommand(Object(Magento\Setup\Console\Command\DiCompileCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#12 /data/vendor/magento/framework/Console/Cli.php(118): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#13 /data/vendor/symfony/console/Application.php(175): Magento\Framework\Console\Cli->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#14 /data/bin/magento(23): Symfony\Component\Console\Application->run()
#15 {main}
```

**Comment by Webwijs** [comment]
I just encountered exactly the same. It seems to depend on the MageWorx\SeoBase extension which seems to be paid.

I guess this was an attempt to make it compatible with the MageWorx SeoSuite, but actually made it incompatible with stores that don't have that extension installed.
`<preference for="MageWorx\SeoBase\Helper\StoreUrl" type="CustomGento\DefaultStoreCodeRemover\Helper\StoreUrl"/>`

**Comment by sprankhub** [comment]
Sorry for the inconvenience! This was discussed in https://github.com/customgento/module-default-store-code-remover-m2/pull/9, but has not been fixed since then. We'll work on it.

**Comment by sprankhub** [comment]
This should be fixed in [v1.0.5](https://github.com/customgento/module-default-store-code-remover-m2/releases/tag/v1.0.5). Sorry again!

**Comment by jbrada** [comment]
Thanks @sprankhub :-)
